cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -111,6 +115,14 @@ public static void main(final String[] args) {
                 writeJSONfile(bom, fileName);
                 break;
 
+            case "signSBOM":
+                bom = signBom();
+                JSONObjectWriter writer = new JSONObjectWriter();
```

Comment:
The writer needs to be constructed from the current SBOM data, you then sign it, before re-writing the signed data back to the output file.
This looks like a full example: https://github.com/cyberphone/openkeystore/blob/0213dd791da447840bb95fb6dfff3e5e7dfccf70/jsf-lab/src/org/webpki/webapps/jsf_lab/JSFService.java#L166